src/test/java/com/karankumar/bookproject/backend/service/BookServiceTest.java

```diff
@@ -86,6 +91,42 @@ public void whenTryingToSaveBookWithoutAuthorExpectNoSave() {
         Assertions.assertEquals(0, bookService.count());
     }
 
+    /**
+     * Tests book is not saved with numberOfPages > max_pages
+     */
+    @Test
+    public void whenTryingToSaveBookWithMaxNumberPageExceedNoSave() {
+        Book book = new Book("Book without author", new Author("First", "Last"), toRead);
+        book.setNumberOfPages(MAX_PAGES + 1);
+        Exception exception  = Assertions.assertThrows(RuntimeException.class, () -> bookService.save(book));
```

Comment:
Its better but would be better if we can try to make it bit more simple what I had suggested in my original comment like:

`Throwable error = Assertions.catchThrowable(() -> bookService.save(book))`

Reply:
The issue was Assertions is already imported from the different package and I have to use a fully qualified package name for it to be used like - 
`org.assertj.core.api.Assertions.catchThrowable()`

That's why I wrote the code like that, if you feel this is okay I will change

Reply:
Ah okay, I got it now..cool, that's fine then.